reflect_abi_patch.go

```diff
@@ -0,0 +1,117 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"maps"
+	"os"
+	"slices"
+	"strings"
+)
+
+func abiNamePatch(path string) (string, error) {
+	data, err := os.ReadFile(path)
+	if err != nil {
+		return "", err
+	}
+
+	find := `return unsafe.String(n.DataChecked(1+i, "non-empty string"), l)`
+	replace := `name := unsafe.String(n.DataChecked(1+i, "non-empty string"), l)
+	return _realName(name)`
+
+	str := strings.Replace(string(data), find, replace, 1)
+
+	realname := `
+//go:linkname _realName
+func _realName(name string) string
+`
+
+	return str + realname, nil
+}
+
+var patchMainFile = ""
+
+func reflectMainPrePatch(path string) ([]byte, error) {
+	if patchMainFile != "" {
+		// already patched another file in main
+		return nil, nil
+	}
+
+	content, err := os.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+
+	patchMainFile = path
+
+	nameMap := "\nvar _nameMap = map[string]string{}"
```

Comment:
why are you adding the empty map in the "pre" step, and replacing it with the filled in map in the "post" step? why not add the filled in map directly in the "post" step?

or, put another way, why do we need separate "pre" and "post" steps? all of this needs a bit more documentation to explain what's going on